### PR TITLE
fixes #2931 - API v2 hosts JSON response, add *_name for each *_id

### PR DESCRIPTION
**Pull request by isratrade**
NOTE: documentation or apipie params was not updated, since the rabl templates are interim and will change soon.

### CODE REVIEW

**Comment by isratrade** [comment]
@mbacovsky, @tstrachota, can you review commit 2011a6e.  It's pretty simple, just adding *_name for each *_id.  
I decided not to add *_names for each *_ids as I prefer show has_many associations in a child array that extends /v2/object_name/base.json.rabl (not done yet)

**Comment by mbacovsky** [comment]
user/index and user/show looks like a good place for inheritance, other than that :+1:

**Comment by isratrade** [comment]
@mbacovsky, I plan to do a lot of re-factoring and DRYing to consolidate to main, base, and show templates, but I wanted to get this merged first so you aren't delayed.  Is hammer using the api v2 responses?

**Comment by mbacovsky** [comment]
@isratrade then i'm okay with the _names part. And yes, hammer is using v2

**Comment by thomasmckay** [comment]
operatingsystem_name is not very useful, is it? It is currently displayed in UI as "RHEL 6.4" (name major.minor). Anyone consuming an api result w/ operatingsystem_id is probably going to make an additional call to get all the details to build out a readable name.

Other than that, looks good. What is the schedule for this merge? (I'd love to use it today! :)

**Comment by mbacovsky** [comment]
I hope we keep operatingsystem_name, it saves hammer a lot of requests...

**Comment by isratrade** [comment]
@mbacovsky, this PR should be merged shortly after #PR 952 is merged

File: app/views/api/v2/hosts/show.json.rabl [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
 if SETTINGS[:locations_enabled]
-  attribute :location_id
-end
-
-child :environment do
-  attributes :id, :name
+  attributes :location_id, :location_name
 end
 
 child :host_parameters do
```
Comment (domcleal) [inline app/views/api/v2/hosts/show.json.rabl]: object_root => false too?

**Comment by domcleal** [comment]
:+1: looks good generally.  I see you made one change to user roles to not use the child root which looks sensible, but now I look, I see lots more cases where this would need changing too.  Do you want to defer that or change them all?

File: app/views/api/v2/hosts/show.json.rabl (hunk 1 above)
Comment (isratrade) [inline app/views/api/v2/hosts/show.json.rabl]: @domcleal, good catch

**Comment by isratrade** [comment]
@domcleal, I will look for child nodes that need changes in this commit.

**Comment by isratrade** [comment]
re-commited with new commit fixes #3768 for child nodes :object_root => false

File: app/views/api/v2/bookmarks/show.json.rabl [hunk 2]
```diff
@@ -1,2 +1,2 @@
 object @bookmark
-attributes :name, :controller, :query, :public, :id, :owner_id, :owner_type
\ No newline at end of file
+extends "api/v2/bookmarks/show"
```
Comment (domcleal) [inline app/views/api/v2/bookmarks/show.json.rabl]: Oops?  Causes a stack overflow

File: app/views/api/v2/config_templates/show.json.rabl [hunk 3]
```diff
@@ -4,7 +4,7 @@ attributes :name, :template, :snippet, :audit_comment, :id
 
 node do |ct|
   unless ct.template_kind.nil?
-    child :template_kind do
+    child :template_kind, :object_root => false do
```
Comment (domcleal) [inline app/views/api/v2/config_templates/show.json.rabl]: Template kind shouldn't be a child node as it's not a has_many relationship

File: app/views/api/v2/bookmarks/show.json.rabl (hunk 2 above)
Comment (isratrade) [inline app/views/api/v2/bookmarks/show.json.rabl]: again, I shouldn't try (what I think is) quick cosmetic fixes :)  I'll revert back to what it was

**Comment by isratrade** [comment]
re-committed with v2/config_templates/show.json.rabl refactored across both commits in PR.

**Comment by domcleal** [comment]
Thanks @isratrade, merged as 636e983 and 1145001 for Foreman 1.4.0.

**Pull request closed without merging**